python/lsst/obs/base/gen2to3/translators.py

```diff
@@ -360,12 +334,105 @@ def addRule(cls, handler: KeyHandler, instrument: Optional[str] = None,
         # find the rules for this instrument, or if we haven't seen it before,
         # add a nested dictionary that matches any DatasetType name and then
         # append this rule.
-        rulesForInstrument = cls._rules.setdefault(instrument, {None: []})
+        rulesForInstrument = self._rules.setdefault(instrument, {None: []})
         rulesForInstrumentAndDatasetType = rulesForInstrument.setdefault(datasetTypeName, [])
         rulesForInstrumentAndDatasetType.append((frozenset(gen2keys), handler, consume))
 
-    @classmethod
-    def makeMatching(cls, datasetTypeName: str, gen2keys: Dict[str, type], instrument: Optional[str] = None,
+    def _addDefaultRules(self):
+        """Add translator rules that should always be present, and don't depend
+        at all on the instrument whose datasets are being converted.
+
+        This is called by `TranslatorFactory` construction.
+        """
+        # Add "skymap" to Gen3 ID if Gen2 ID has a "tract" key.
+        self.addRule(SkyMapKeyHandler(), gen2keys=("tract",), consume=False)
+
+        # Add "skymap" to Gen3 ID if DatasetType is one of a few specific ones
+        for coaddName in ("deep", "goodSeeing", "psfMatched", "dcr"):
+            self.addRule(SkyMapKeyHandler(), datasetTypeName=f"{coaddName}Coadd_skyMap")
+
+        # Translate Gen2 str patch IDs to Gen3 sequential integers.
+        self.addRule(PatchKeyHandler(), gen2keys=("patch",))
+
+        # Copy Gen2 "tract" to Gen3 "tract".
+        self.addRule(CopyKeyHandler("tract", dtype=int), gen2keys=("tract",))
+
+        # Add valid_first, valid_last to instrument-level transmission/ datasets;
+        # these are considered calibration products in Gen3.
+        for datasetTypeName in ("transmission_sensor", "transmission_optics", "transmission_filter"):
+            self.addRule(ConstantKeyHandler("calibration_label", "unbounded"),
+                         datasetTypeName=datasetTypeName)
+
+        # Translate Gen2 pixel_id to Gen3 skypix.
+        # TODO: For now, we just assume that the refcat indexer uses htm7, since that's
```

Comment:
Is this really a TODO, or is it a NOTE? (I know it was carried over, but there's not an obvious ticket that could be filed about it now, so it doesn't seem like something that could be "TO-DONE")

Reply:
Well, it's a problem I think we're going to have to solve in the not-terribly-distant future, even though I don't know how I'd go about it and therefore have not been eager to describe it in a ticket.  I suppose I find TODO vs. NOTE a good way to tell some poor soul who runs into the problem that it's reasonable for them to complain and get the feature implemented, rather than resign themselves to working around it.

Reply:
That's a fair point. In that case, can we make that more explicit with `TODO (no implementation ticket, but complain to #dm-middleware if this becomes a problem)` or something? I have now been trained to want a ticket whenever I see a TODO.